packages/react/src/components/CreateConsumer.tsx

```diff
@@ -0,0 +1,86 @@
+import { useContext, useState } from "react";
+import { ErrorContext } from "./ConsumerControl";
+import { useDataContext, useProviderContext } from "./context";
+import { CheckIcon, Spinner } from "./icons";
+
+export default function CreateConsumer() {
+  const provider = useProviderContext();
+  const [, setError] = useContext(ErrorContext);
+  const [dataModel, setDataModel] = useDataContext();
+  const [editMode, setEditMode] = useState<boolean>(false);
+  const [label, setLabel] = useState<string>("");
+  const [isCreating, setIsCreating] = useState<boolean>(false);
+  const [isInvalid, setIsInvalid] = useState<boolean>(false);
+
+  async function handleCreateConsumer() {
+    try {
+      if (label.trim().length === 0) {
+        setIsInvalid(true);
+        return;
+      }
+      setIsInvalid(false);
+
+      if (!provider.createConsumer) {
+        throw new Error(
+          "Provider does not implement createConsumer but enableCreateConsumer is true",
+        );
+      }
+
+      setIsCreating(true);
+      await provider.createConsumer(label);
+      const result = await provider.getConsumers();
+      setDataModel({ ...dataModel, consumers: result.data });
+      setEditMode(false);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
+      setError(err.message);
+    } finally {
+      setIsCreating(false);
+    }
+  }
+
+  function handleLabelChange(e: React.ChangeEvent<HTMLInputElement>) {
+    if (e.target.value.trim().length > 0) {
+      setIsInvalid(false);
+    }
+    setLabel(e.target.value);
+  }
+
+  if (!editMode) {
+    return (
+      <button
+        onClick={() => setEditMode(true)}
+        className="text-white rounded bg-pink-500 hover:bg-pink-700 p-2 px-4"
+      >
+        Create new API Key
+      </button>
+    );
+  }
+
+  return (
+    <div className="flex flex-row items-center gap-x-2 w-full">
```

Comment:
No inline tailwind anymore - you have to do it through a css file like the other files do. Its a bit tedious to do so I usually ask chatGPT to extract them for me :)